src/lib/server/websearch/generateQuery.ts

```diff
@@ -1,46 +1,68 @@
 import type { Message } from "$lib/types/Message";
 import { format } from "date-fns";
 import { generateFromDefaultEndpoint } from "../generateFromDefaultEndpoint";
-import { defaultModel } from "../models";
+import { smallModel } from "../models";
 
 export async function generateQuery(messages: Message[]) {
 	const currentDate = format(new Date(), "MMMM d, yyyy");
 	const userMessages = messages.filter(({ from }) => from === "user");
 	const previousUserMessages = userMessages.slice(0, -1);
+
 	const lastMessage = userMessages.slice(-1)[0];
-	const promptSearchQuery = defaultModel.webSearchQueryPromptRender({
-		message: lastMessage,
-		previousMessages: previousUserMessages.map(({ content }) => content).join(" "),
-		currentDate,
-	});
-	const searchQuery = await generateFromDefaultEndpoint(promptSearchQuery).then((query) => {
-		// example of generating google query:
-		// case 1
-		// user: tell me what happened yesterday
-		// LLM: google query is "news september 12, 2023"
-		// the regex below will try to capture the last "phrase" (i.e. words between quotes or double quotes or ticks)
-		// in this case, it is "news september 12, 2023"
-		// if there is no "phrase", we will just use the user query, which was "tell me what happened yesterday"
-		const regexLastPhrase = /("|'|`)((?:(?!\1).)+)\1$/;
-		let match = query.match(regexLastPhrase);
-		if (match) {
-			return match[2];
-		}
 
-		// case 2
-		// user: tell me what happened yesterday
-		// LLM: Here is a query: news september 12, 2023
-		// the regex below will try to capture the last sentences starting from :
-		// in this case, it is "news september 12, 2023"
-		// if there is no math, we will just use the user query, which was "tell me what happened yesterday"
-		const regexColon = /:\s(.*)$/;
-		match = query.match(regexColon);
-		if (match) {
-			return match[1];
-		}
+	const convQuery: Array<Omit<Message, "id">> = [
+		{
+			from: "user",
+			content: `Previous Questions:
+- Who is the president of France?
+
+Current Question: What about Mexico?
+`,
+		},
+		{
+			from: "assistant",
+			content: "President of Mexico",
+		},
+		{
+			from: "user",
+			content: `Previous questions: 
+- When is the next formula 1 grand prix?
+
+Current Question: Where is it being hosted ?`,
+		},
+		{
+			from: "assistant",
+			content: "location of next formula 1 grand prix",
+		},
+		{
+			from: "user",
+			content: "Current Question: What type of printhead does the Epson F2270 DTG printer use?",
+		},
+		{
+			from: "assistant",
+			content: "Epson F2270 DTG printer printhead",
+		},
+		{ from: "user", content: "What were the news yesterday ?" },
+		{ from: "assistant", content: `news ${new Date(Date.now() - 864e5).toDateString()}` },
```

Comment:
Oooh that's better formatting indeed

Reply:
please feel free to merge this commit suggestion whenever you want 👍

Reply:
I did it [slightly different](https://github.com/huggingface/chat-ui/blob/1f01e9b7f175308365725b899c377c9c354f1c0c/src/lib/server/websearch/generateQuery.ts#L48-L49) because I wantes yesterday not currentDate but I reused the date format!